aptos-move/aptos-aggregator/src/resolver.rs

```diff
@@ -18,29 +28,89 @@ pub enum AggregatorReadMode {
 /// Returns a value of an aggregator from cache or global storage.
 ///   - Ok(..)       if aggregator value exists
 ///   - Err(..)      otherwise.
-pub trait AggregatorResolver {
+pub trait TAggregatorView {
+    type Identifier;
+
+    fn get_aggregator_v1_state_value(
+        &self,
+        id: &Self::Identifier,
+    ) -> anyhow::Result<Option<StateValue>>;
```

Comment:
Why Reslut<Option<..>>? If aggregator value exists, can the aggregator value still be None?

Reply:
Yes, actually a good point to discuss: before we mentioned that V1 can never be deleted but if we upgrade and transition to V2, V1 would follow the same flow with replacement (internally). Then there is a question if we keep it like this forever or get rid of this state item. Then deletion can happen?

Reply:
Let's allow deletion for V1, however, we'd need to test it well before we deprecate / delete. We can decide to not delete at any time as well anyway.

Reply:
I am fine with leaving any migration/replacement implementation changes for later - we will need to test those out well anyways

Reply:
Returned option for now, let's revisit this later once we consider migration